mettle/src/stdapi/sys/memory.c

```diff
@@ -479,6 +479,7 @@ struct addr_range *parse_maps_file(pid_t pid)
 			struct addr_range *range = malloc(sizeof(struct addr_range));
 			if(range == NULL)
 			{
+				fclose(fp);
```

Comment:
Isn't this also missing a `free(line)` and a `regfree(&regex)`?

Reply:
Oops, yep!